packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -318,7 +318,7 @@ export class UserPool extends cdk.Construct implements IUserPool {
    */
   public readonly userPoolProviderUrl: string;
 
-  private triggers: CfnUserPool.LambdaConfigProperty = { };
+  private triggers: Mutable<CfnUserPool.LambdaConfigProperty> = { };
```

Comment:
Interesting trick, but I actually found it refreshing to rewrite some of the code to rely on an the structure being immutable. The results were usually better and more readable code.

I am wondering also how this creeped up here.

Reply:
Yeah this was actually not desired here... I'm looking to remove those spurious changes from the PR.